AmplifyPlugins/Analytics/AWSPinpointAnalyticsPlugin/LocalStorage/SQLiteLocalStorageAdapter.swift

```diff
@@ -0,0 +1,136 @@
+//
+// Copyright Amazon.com Inc. or its affiliates.
+// All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+import Foundation
+import SQLite
+
+/// Local storage adapter that implements local storage using SQLite.swift
+final class SQLiteLocalStorageAdapter: LocalStorageProtocol {
+    private var connection: Connection?
+    private var dbFilePath: URL?
+    
+    /// Initializer
+    /// - Parameter databaseName: The database name
+    convenience init(databaseName: String) throws {
+        var dbFilePath = SQLiteLocalStorageAdapter.getDbFilePath(databaseName: databaseName)
+        let connection: Connection
+        do {
+            connection = try Connection(dbFilePath.absoluteString)
+            var urlResourceValues = URLResourceValues()
+            urlResourceValues.isExcludedFromBackup = true
+            try dbFilePath.setResourceValues(urlResourceValues)
+        } catch {
+            throw LocalStorageError.invalidDatabase(path: dbFilePath.absoluteString, error)
+        }
+
+        try self.init(connection: connection, dbFilePath: dbFilePath)
+    }
+    
+    /// Initializer
+    /// - Parameters:
+    ///   - connection: SQLite Connection
+    ///   - dbFilePath: Path to the database
+    private init(
+        connection: Connection,
+        dbFilePath: URL? = nil
+    ) throws {
+        self.connection = connection
+        self.dbFilePath = dbFilePath
+        try initializeDatabase(connection: connection)
+    }
+    
+    /// Create the Event and Dirty Event Tables
+    private func createTables() throws {
+        guard let connection = connection else {
+            throw LocalStorageError.invalidOperation(causedBy: nil)
+        }
+
+        let createEventTableStatement = """
+            CREATE TABLE IF NOT EXISTS Event (
```

Comment:
I'm wondering if these table definitions should be owned by this adapter, or if they should be provided by its consumer, i.e. the `AnalyticsEventStorage`

Reply:
That's a good suggestion, I've moved all query statements that are specific to events out of the adapter and into the storage class.